src/core/reducers/data/personalization.ts

```diff
@@ -0,0 +1,58 @@
+import { createTransform, persistReducer } from 'redux-persist';
+import storage from 'redux-persist/lib/storage';
+import Actions from '../../actions';
+import Adapter from '../../adapters/personalization';
+import Selectors from '../../selectors';
+import Store from '../../store';
+
+export type Action = Actions.UpdateBiasing;
+export type State = Store.Personalization;
+
+export const STORAGE_KEY = 'gb-personalization';
+export const STORAGE_WHITELIST = ['biasing'];
+
+export const DEFAULT: State = {
+  biasing: {
+    allIds: [],
+    byId: {},
+  }
+};
+
+function updatePersonalization(state: State = DEFAULT, action: Action): State {
+  switch (action.type) {
+    case Actions.UPDATE_BIASING: return updateBiasing(state, action.payload);
+    default: return state;
+  }
+}
+
+export const updateBiasing = (state: State, payload: Actions.Payload.Personalization.Biasing) => {
+  const byId = {
+    ...state.biasing.byId,
+    [payload.variant]: {
+      ...state.biasing.byId[payload.variant],
+      [payload.key]: payload.bias
+    }
+  };
+
+  let allIds = insertSorted(state.biasing.allIds, { variant: payload.variant, key: payload.key });
+  const config = payload.config;
+  allIds = Adapter.pruneBiases(allIds, payload.variant, Object.keys(byId[payload.variant]).length, config);
```

Comment:
I'd rather not re-assign this value right after initializing it, maybe return the value of `pruneBiases()` inline in the returned object literal